pkg/segment/structs/evaluationstructs.go

```diff
@@ -2017,7 +2062,9 @@ func (self *NumericExpr) Evaluate(fieldToValue map[string]utils.CValueEnclosure)
 			}
 			number, err := strconv.ParseInt(strValue, base, 64)
 			if err != nil {
-				return 0, fmt.Errorf("NumericExpr.Evaluate: cannot convert '%v' to number with base %d", strValue, base)
+				// DO NOT return an error if the string is not a number
+				// just return 0
+				return 0, nil
```

Comment:
This seems a little odd. Why should we return a nil error?

Reply:
Same as above, if we `tonumber(string)`, splunk returns `0`.